### PR TITLE
Chore/modify recipe

### PR DESCRIPTION
**Pull request by haybankz**
Remove duplicate data in recipe object

### CODE REVIEW

File: lib/utils/utils.dart [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import 'dart:io';
+
+import 'package:easel_flutter/main.dart';
+import 'package:flutter/material.dart';
+import 'package:url_launcher/url_launcher.dart';
+
+void showAlertDialog(String message, {Widget button = const SizedBox()}) {
```
Comment (kjawadDeveloper2) [inline lib/utils/utils.dart]: Make it a class object we dont need it as a global thing.

File: lib/utils/utils.dart [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        child: AlertDialog(
+          content: Text(message),
+          actions: [button],
+        ),
+      ));
+}
+
+void launchAppStore()async{
+  if(Platform.isAndroid){
+    const  _url = "https://play.google.com/store/apps/details?id=tech.pylons.wallet";
```
Comment (kjawadDeveloper2) [inline lib/utils/utils.dart]: Put this url in constant file.

File: lib/screens/description_screen.dart [hunk 3]
```diff
@@ -67,6 +67,8 @@ class _DescriptionScreenState extends State<DescriptionScreen> {
                         textCapitalization: TextCapitalization.sentences,
                         validator: (value) {
                           if (value!.isEmpty) return "Enter NFT name";
+                          if(value.length <= kMinNFTName) return "NFT name should have $kMinNFTName characters or more";
```
Comment (kjawadDeveloper2) [inline lib/screens/description_screen.dart]: Move strings to a constant file.

File: lib/screens/routing_screen.dart [hunk 4]
```diff
@@ -81,11 +80,10 @@ class _RoutingScreenState extends State<RoutingScreen> {
...
               children: [
                 ValueListenableBuilder<String>(
-                  valueListenable: username,
-                  builder: (_, String name, __) {
-                    return Text("Welcome to Easel, $name");
-                  }
-                ),
+                    valueListenable: username,
+                    builder: (_, String name, __) {
+                      return Text("Welcome to Easel, $name");
```
Comment (kjawadDeveloper2) [inline lib/screens/routing_screen.dart]: Move text to constant file.

File: lib/utils/amount_formatter.dart [hunk 5]
```diff
@@ -3,25 +3,29 @@ import 'package:intl/intl.dart';
...
       return newValue;
     }
     if (newValue.selection.baseOffset > maxDigits) {
       return oldValue;
     }
     double value = double.parse(newValue.text);
-    final formatter =  NumberFormat("#,###", "en_US");
-    String newText = formatter.format(value);
+    final formatter =  NumberFormat(isDecimal ? "#,##0.00" : "#,###", "en_US");
+    // formatter = NumberFormat("#,##0.00", "en_US");
```
Comment (kjawadDeveloper2) [inline lib/utils/amount_formatter.dart]: Remove dead code please.

File: .flutter-plugins [hunk 6]
```diff
@@ -14,7 +14,7 @@ shared_preferences_windows=/Users/mac/development/flutter/.pub-cache/hosted/pub.
 sqflite=/Users/mac/development/flutter/.pub-cache/hosted/pub.dartlang.org/sqflite-2.0.0+4/
 uni_links=/Users/mac/development/flutter/.pub-cache/hosted/pub.dartlang.org/uni_links-0.5.1/
 uni_links_web=/Users/mac/development/flutter/.pub-cache/hosted/pub.dartlang.org/uni_links_web-0.1.0/
-url_launcher=/Users/mac/development/flutter/.pub-cache/hosted/pub.dartlang.org/url_launcher-6.0.12/
```
Comment (kjawadDeveloper2) [inline .flutter-plugins:17]: Add this file in gitgnore and remove it from the remote .

File: lib/screens/description_screen.dart [hunk 7]
```diff
@@ -66,9 +66,12 @@ class _DescriptionScreenState extends State<DescriptionScreen> {
                         controller: provider.artNameController,
                         textCapitalization: TextCapitalization.sentences,
                         validator: (value) {
-                          if (value!.isEmpty) return "Enter NFT name";
-                          if(value.length <= kMinNFTName) return "NFT name should have $kMinNFTName characters or more";
-
+                          if (value!.isEmpty) {
+                            return "Enter NFT name";
```
Comment (kjawadDeveloper2) [inline lib/screens/description_screen.dart]: Move strings to constants.

**Review by kjawadDeveloper2 — CHANGES_REQUESTED** [review]
Kindly make these changes please.

**Review by mustafapylons — APPROVED** [review]
Looks good after the requested changes are done.

**Review by kjawadDeveloper1 — APPROVED** [review]
Looks Good from my side.

**Review by kjawadDeveloper2 — APPROVED** [review]
Looks good from my side.

**Pull request merged**